conda/core/portability.py

```diff
@@ -28,7 +28,7 @@
                  br'(.*)'  # the rest of the line can contain option flags
                  br')$')  # end whole_shebang group
 
-MAX_SHEBANG_LENGTH = 127 if on_linux else 512  # Not used on Windows
+MAX_SHEBANG_LENGTH = 127 if (on_linux or on_mac) else 512  # Not used on Windows
```

Comment:
Flip it?

```suggestion
MAX_SHEBANG_LENGTH = 512 if on_win else 127
```

Reply:
Yeah, that's easy to parse! (for humans)